src/Blocks/BlockBigFlower.h

```diff
@@ -77,7 +77,15 @@ class cBlockBigFlowerHandler :
 
 	virtual bool CanBeAt(cChunkInterface & a_ChunkInterface, int a_RelX, int a_RelY, int a_RelZ, const cChunk & a_Chunk) override
 	{
-		return ((a_RelY > 0) && (a_Chunk.GetBlock(a_RelX, a_RelY - 1, a_RelZ) != E_BLOCK_AIR) && (a_RelY < cChunkDef::Height - 1) && ((a_Chunk.GetBlock(a_RelX, a_RelY + 1, a_RelZ) == E_BLOCK_AIR) || (a_Chunk.GetBlock(a_RelX, a_RelY + 1, a_RelZ) == E_BLOCK_BIG_FLOWER)));
+		if (a_RelY <= 0)
+		{
+			return false;
+		}
+		BLOCKTYPE BlockType;
+		NIBBLETYPE BlockMeta;
+		a_Chunk.GetBlockTypeMeta(a_RelX, a_RelY - 1, a_RelZ, BlockType, BlockMeta);
+
+		return IsBlockTypeOfDirt(BlockType) || ((BlockType == E_BLOCK_BIG_FLOWER) && !(BlockMeta & 0x08));
```

Comment:
Style: Replace `!(BlockMeta & 0x08)` with an explicit comparison - we only allow bool values in conditions, the result of that expression is NOT a bool value.

Perhaps even better, replace it with a call to new function `IsMetaTopPart` - adds proper meaning and is much more maintainable.